### PR TITLE
Add namespace metadata configuration document

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/16209

### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [x] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/master/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

File: src/content/app-platform/defaulting-validation/index.md [hunk 1]
```diff
@@ -111,6 +111,7 @@ Currently we validate:
 
 - The `app-operator.giantswarm.io/version` label is present.
 - All referenced configmaps and secrets exist.
+- More than one App CRs would not try to update the same label or annotation as a different value.
```
Comment (tomahawk28) [inline src/content/app-platform/defaulting-validation/index.md:114]: Updating the validation doc as well

File: src/content/app-platform/namespace-configuration/index.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+---
+linkTitle: App namespace metadata
+title: App's namespace metadata
```
Comment (cokiengchiara) [inline src/content/app-platform/namespace-configuration/index.md:3]:
```suggestion
title: How to annotate an app's namespace via its App CR
```

Comment (cokiengchiara) [reply src/content/app-platform/namespace-configuration/index.md:3]: Want to make it clearer in the headline what this helps the user to do.

File: src/content/app-platform/namespace-configuration/index.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+aliases:
+  - /reference/app-configuration/
+user_questions:
+  - How can I label the app CR's namespace?
+  - How can I annotate the app CR's namespace?
+---
+
+# App CR's namespace configuration
+
+At some point, you might want to put the specific labels or annotations to the app's namespace to bring benefits from `linkerd` or `kiam`.
```
Comment (cokiengchiara) [inline src/content/app-platform/namespace-configuration/index.md:20]:
```suggestion
At some point, you might want to put the specific labels or annotations to the app's namespace to bring benefits from `linkerd` or `kiam`. For example, to include everything in a namespace to the Linkerd service mesh. Or, to tell KIAM you do allow Loki to get logs about pods running in this namespace. 
```

Maybe add example to illustrate? Is this correct?

File: src/content/app-platform/namespace-configuration/index.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  namespace: kiam
+  namespaceConfig:
+    labels:
+      monitoring: "enabled"
+```
+
+### Validation in App CR's Namespace Config
+
+In [default validation logic]({{< relref "/app-platform/defaulting-validation" >}}), app-admission-controller checks whether multiple app CRs are updating the same namespace
+with different values. This logic would avoid the situation where the same namespace keeps changing at each app CR's reconciliation.
```
Comment (cokiengchiara) [inline src/content/app-platform/namespace-configuration/index.md:78]:
```suggestion
with different values. If that is the case, it prompts the user to confirm which namespace to use. This logic would avoid the situation where the same namespace keeps changing at each app CR's reconciliation.
```

What does app-admission-controller do in the case where multiple app CRs are updating the same namespace w different values? I just wrote some BS that is probably wrong. Haha.

File: src/content/app-platform/defaulting-validation/index.md (hunk 1 above)
Comment (cokiengchiara) [reply src/content/app-platform/defaulting-validation/index.md:114]: perfect thanks!

**Comment by cokiengchiara** [comment]
Have few comments but LGTM thanks @tomahawk28 ❤️

File: src/content/app-platform/namespace-configuration/index.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+---
+
+# App CR's namespace configuration
+
+At some point, you might want to put the specific labels or annotations to the app's namespace to bring benefits from `linkerd` or `kiam`.
+This feature enables you to add labels and annotations to the app's namespace by updating [App CRs]({{< relref "/ui-api/management-api/crd/apps.application.giantswarm.io.md" >}}).
+
+## Configuring the App CR's namespace
+
+### Updating labels/annotations
```
Comment (rossf7) [inline src/content/app-platform/namespace-configuration/index.md:25]:
```suggestion
## Configuring labels / annotations
```

I think we can simplify to have less headings.

**Review by rossf7 — COMMENTED** [review]
@tomahawk28 Thanks for adding and the examples are great.

One thing I forgot but that I think will help is if we call it the target namespace. helm-controller uses this in its helmrelease CRDs and I think its clearer.

File: src/content/app-platform/namespace-configuration/index.md (hunk 3 above)
Comment (tomahawk28) [reply src/content/app-platform/namespace-configuration/index.md:20]: Perfect, thanks.

File: src/content/app-platform/namespace-configuration/index.md (hunk 4 above)
Comment (tomahawk28) [reply src/content/app-platform/namespace-configuration/index.md:78]: It's not BS! :) I think we can say `If this is the case, it rejects entries of the latest app CR`

File: src/content/app-platform/namespace-configuration/index.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+spec:
+  catalog: giantswarm
+  name: kiam
+  namespace: kiam
+  namespaceConfig:
+    labels:
+      monitoring: "enabled"
+```
+
+### Validation in App CR's Namespace Config
```
[the remark this answers is not in the document]
Comment (tomahawk28) [reply src/content/app-platform/namespace-configuration/index.md:75]: Much better, thanks.

File: src/content/app-platform/namespace-configuration/index.md (hunk 4 above)
Comment (tomahawk28) [reply src/content/app-platform/namespace-configuration/index.md:78]: I would like to use Ross's suggestion here, https://github.com/giantswarm/docs/pull/901#discussion_r613106463 
Using the word `prevents` makes sense to me.

File: src/content/app-platform/namespace-configuration/index.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-batman
+aliases:
+  - /reference/app-configuration/
+user_questions:
+  - How can I label the app CR's namespace?
+  - How can I annotate the app CR's namespace?
+---
+
+# App CR's namespace configuration
```
[the remark this answers is not in the document]
Comment (tomahawk28) [reply src/content/app-platform/namespace-configuration/index.md:18]: To follow the defaulting validation logic document, I think it makes sense to add it as below. 
```
# App CR's target namespace configuration

## Overview
```
Right?

File: src/content/app-platform/namespace-configuration/index.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+title: How to annotate an app's target namespace via its App CR
+description: Configuring metadata for the target namespace of an app via its app CR so it can be used by other tools such as service meshes.
+weight: 20
+menu:
+  main:
+    parent: app-platform
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-batman
+aliases:
+  - /reference/app-configuration/
```
Comment (rossf7) [inline src/content/app-platform/namespace-configuration/index.md:12]:
```suggestion
```

File: src/content/app-platform/namespace-configuration/index.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+user_questions:
+  - How can I label an app CR's target namespace?
+  - How can I annotate an app CR's target namespace?
+---
+
+# App CR's target namespace configuration
+
+## Overview
+
+At some point, you might want to put the specific labels or annotations to the app's namespace to bring benefits from `linkerd` or `kiam`. For example, to include everything in a namespace to the Linkerd service mesh. Or, to tell KIAM you do allow Loki to get logs about pods running in this namespace.
```
Comment (rossf7) [inline src/content/app-platform/namespace-configuration/index.md:22]:
```suggestion
At some point, you might want to add specific labels or annotations to an app's target namespace to enable other tools such as `linkerd` or `kiam`. For example, to include all pods in a namespace in the `linkerd` service mesh. Or, to tell `kiam` you do allow Loki to get logs from pods running in this namespace.
```

File: src/content/app-platform/namespace-configuration/index.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+This feature enables you to add labels and annotations to the app's namespace by updating [App CRs]({{< relref "/ui-api/management-api/crd/apps.application.giantswarm.io.md" >}}).
```
Comment (rossf7) [inline src/content/app-platform/namespace-configuration/index.md:23]:
```suggestion
This feature enables you to add labels and annotations to the app's target namespace via its [App CR]({{< relref "/ui-api/management-api/crd/apps.application.giantswarm.io.md" >}}). This is simpler than using other solutions such as triggering a Kubernetes jpb via a Helm hook.
```

I think we need to describe the alternative so its clear why we built this.

File: src/content/app-platform/namespace-configuration/index.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+## Configuring labels / annotations
+
+You can use `.spec.namespaceConfig` to configure the namespace metadata. You can either choose `.spec.namespaceConfig.annotations` or
+`spec.namespaceConfig.labels` and provide the values as a key/value map. App platform then ensures the namespace exists and has this metadata.
```
Comment (rossf7) [inline src/content/app-platform/namespace-configuration/index.md:28]:
```suggestion
You can use `.spec.namespaceConfig` to configure the namespace metadata. You can set `.spec.namespaceConfig.annotations` and / or
`spec.namespaceConfig.labels` and provide the values as a key/value map. App platform then ensures the namespace exists and has this metadata.
```

**Review by rossf7 — COMMENTED** [review]
This is looking good. Just a final set of suggestions from me.

File: src/content/app-platform/namespace-configuration/index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (rossf7) [reply src/content/app-platform/namespace-configuration/index.md]: Yes I missed that. Thanks good to keep aligned with the other page. 👍

File: src/content/app-platform/namespace-configuration/index.md (hunk 8 above)
Comment (tomahawk28) [reply src/content/app-platform/namespace-configuration/index.md:12]: 🙈 Sorry, it was left over from copy & paste.

File: src/content/app-platform/namespace-configuration/index.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
+---
+linkTitle: App target namespace metadata
+title: How to annotate an app's target namespace via its App CR
+description: Configuring metadata for the target namespace of an app via its app CR so it can be used by other tools such as service meshes.
+weight: 20
```
Comment (rossf7) [inline src/content/app-platform/namespace-configuration/index.md:5]:
```suggestion
weight: 40
```

This controls the order on https://docs.giantswarm.io/app-platform/ 

This feature is great but only applies to some app CRs. Can you also bump `apps` to 50?

**Review by rossf7 — APPROVED** [review]
LGTM!

Just a final nit on the sort order.

File: src/content/app-platform/namespace-configuration/index.md (hunk 12 above)
Comment (tomahawk28) [reply src/content/app-platform/namespace-configuration/index.md:5]: Thanks, TIL this is ordering by descending order.

**Pull request merged**